### PR TITLE
cam6_3_119: CAM-chem updates

### LINKED ISSUE #538: Issue with maximum number of chemical emissions files #538
The maximum number of emissions files allowed needs to be increased for some cases.  The maximum number is limited by the number of transported tracers and the number of species in the chemical mechanism.  For CAM (without full chemistry) this may be too limited.

### LINKED ISSUE #771: Updates to FCHIST and FCnudged compsets
### What is the feature/what would you like to discuss?

From @tilmes :

1. env_run.xml
SSTs 
we currently have a file that ends in 2021
 <entry id="SSTICE_DATA_FILENAME" value="$DIN_LOC_ROOT/atm/cam/sst/sst_HadOIBl_bc_1x1_1850_2021_c120422.nc">

we also have one that continues till 2023:  (this however is a repeat of 2021 till 2023, and is not really based on observations, therefore probably not desirable for a compset): 
<entry id="SSTICE_DATA_FILENAME" value="/glade/p/acom/acom-climate/shawnh/cesm2_sstice/[sst_HadOIBl_bc_0.9x1.25_1850_2023_c220203.nc](http://sst_hadoibl_bc_0.9x1.25_1850_2023_c220203.nc/)">

Is there a way to change the model so it would take the latest SST year we have and instead of defaulting to 1850 when the file ends? 

Startdate:
The  current startdate is 2010, I am  fine to keep that.. is that OK?

Namelist updates:
Changes needed:
csw_time_type = 'SERIAL'
dust_emis_fact         = 0.38D0

Add to output: 'OCN_FLUX_DMS'

Emissions etc: move to SSP8.5:
see: 
/glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/f.e22.FCnudged.f09_f09_mg17.cam6_3_100_updates.001/user_nl_cam

Volcanoes: Do we want  to have Hunga Tonga as a default in this compset?

### Is there anyone in particular you want to be part of this conversation?

@tilmes, @lkemmons 

### Will this change (regression test) answers?

Yes

### Will you be implementing this enhancement yourself?

Yes, but I will need some help

### PR DESCRIPTION
**Pull request by fvitt**
closes #771 
fixes #538

### CODE REVIEW

**Review by cacraigucar — APPROVED** [review]

File: src/chemistry/mozart/chemistry.F90 [hunk 1]
```diff
@@ -760,53 +726,106 @@ subroutine chem_init(phys_state, pbuf2d)
...
+       call ghg_chem_init(phys_state, bndtvg, h2orates)
+    endif
+
+    call init_cfc11star(pbuf2d)
+
+    ! MEGAN emissions initialize
+    if (shr_megan_mechcomps_n>0) then
+
+       allocate( megan_indices_map(shr_megan_mechcomps_n) )
+       allocate( megan_wght_factors(shr_megan_mechcomps_n) )
```
Comment (nusbaume) [inline src/chemistry/mozart/chemistry.F90:739]: I would add `stat` checks to these allocate statements.

File: src/chemistry/mozart/chemistry.F90 [hunk 2]
```diff
@@ -760,53 +726,106 @@ subroutine chem_init(phys_state, pbuf2d)
...
+
+    logical function aero_has_emis(spcname)
+      use seasalt_model, only: seasalt_names
+      use dust_model, only: dust_names
+
+      character(len=*) :: spcname
+
+      aero_has_emis = any(seasalt_names(:) == spcname).or.any(dust_names(:) == spcname)
+
+    end function aero_has_emis
```
Comment (nusbaume) [inline src/chemistry/mozart/chemistry.F90:828]: I believe this function can be `pure`.

Also, I might add `intent(in)` to the `spcname` variable just to be paranoid.

File: src/chemistry/mozart/ocean_emis.F90 [hunk 3]
```diff
@@ -956,8 +961,30 @@ subroutine cseawater_ini()
...
+          call add_default('OCN_FLUX_' // trim(Csw_nM(m)%species), 1, ' ')
+       end if
     end do
 
   end subroutine cseawater_ini
 
+  !--------------------------------------------------------------------------------
+  ! returns TRUE of species has ocean emissions
+  !--------------------------------------------------------------------------------
+  logical function ocean_emis_species(name)
```
Comment (nusbaume) [inline src/chemistry/mozart/ocean_emis.F90:974]: I believe this function can be `pure`.

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good, but I did have a few small requests and one question.

File: bld/namelist_files/use_cases/hist_trop_strat_vbsfire_cam6.xml [hunk 4]
```diff
@@ -87,104 +87,36 @@
...
-         'soa2_a2_CHML', 'soa3_a2_CHML', 'soa4_a2_CHML', 'soa5_a2_CHML', 'so4_a1_CHMP', 'so4_a2_CHMP', 'so4_a3_CHMP', 'soa1_a1_CHMP', 'soa2_a1_CHMP',
-         'soa3_a1_CHMP', 'soa4_a1_CHMP', 'soa5_a1_CHMP', 'soa1_a2_CHMP', 'soa2_a2_CHMP', 'soa3_a2_CHMP', 'soa4_a2_CHMP', 'soa5_a2_CHMP', 'r_jsoa1_a1',
-         'r_jsoa2_a1', 'r_jsoa3_a1', 'r_jsoa4_a1', 'r_jsoa5_a1', 'r_jsoa1_a2', 'r_jsoa2_a2', 'r_jsoa3_a2', 'r_jsoa4_a2', 'r_jsoa5_a2',
-         'r_GLYOXAL_aer', 'H2SO4_sfnnuc1', 'num_a2_sfnnuc1', 'TMOCS', 'TMSO2', 'TMDMS', 'TMso4_a1', 'TMso4_a2', 'TMso4_a3',
-         'BURDENDUSTdn', 'BURDENPOMdn', 'BURDENSO4dn', 'BURDENSOAdn', 'BURDENSEASALTdn','BURDENBCdn', 'PM25',
-         'O3_Prod','O3_Loss','O3S_Loss','RO2_NO_sum','O3_alkenes','RO2_NO3_sum','RO2_HO2_sum','RO2_RO2_sum','RCO2_NO2_sum',
-         'OddOx_Ox_Loss','OddOx_HOx_Loss','OddOx_NOx_Loss','OddOx_CLOxBROx_Loss','OddOx_Loss_Tot','OddOx_Prod_Tot',
-         'Ox_Prod','Ox_Loss'
+   'ACTREL', 'AQ_SO2', 'AREA', 'BROX', 'BROY', 'BRY', 'CLOX', 'CLOY', 'CLY',  'NOX', 'NOY', 'TBRY', 'TCLY',  'CFC11STAR',
```
[the remark this answers is not in the document]
Comment (fvitt) [reply bld/namelist_files/use_cases/hist_trop_strat_vbsfire_cam6.xml:90]: Yes. Thanks.

File: src/chemistry/mozart/chemistry.F90 (hunk 1 above)
Comment (fvitt) [reply src/chemistry/mozart/chemistry.F90]: done

File: src/chemistry/mozart/chemistry.F90 (hunk 2 above)
Comment (fvitt) [reply src/chemistry/mozart/chemistry.F90]: done

File: src/chemistry/mozart/ocean_emis.F90 (hunk 3 above)
Comment (fvitt) [reply src/chemistry/mozart/ocean_emis.F90]: done

File: src/chemistry/mozart/ocean_emis.F90 [hunk 5]
```diff
@@ -971,7 +971,7 @@ end subroutine cseawater_ini
   !--------------------------------------------------------------------------------
   ! returns TRUE of species has ocean emissions
```
Comment (nusbaume) [inline src/chemistry/mozart/ocean_emis.F90:972]: I assume `returns TRUE of` should be `returns TRUE if`?

**Review by nusbaume — APPROVED** [review]
Looks great now, thanks!  I did find one possible typo in a comment, but I don't think it requires holding up this PR for another review.

**Pull request merged**